awx/main/tasks/system.py

```diff
@@ -240,15 +240,10 @@ def apply_cluster_membership_policies():
         logger.debug('Cluster policy computation finished in {} seconds'.format(time.time() - started_compute))
 
 
-@task(queue='tower_broadcast_all')
+@task(queue='tower_settings_change')
 def clear_setting_cache(setting_keys):
-    orig_len = len(setting_keys)
```

Comment:
i think this still need to be here since redis cache still need to be cleared in the task pods

Reply:
alternatively we just launch cache-clear in both web and task therefore this method would only be there to serve to send pg_notify

Reply:
additional random qustion... if i have 3 task pod would this send out 3 pg_notify or 1?

Reply:
can we just send the pg_notify during when we write setting to the database? this way we completely eliminate the need of even queuing this job

Reply:
I'm pretty sure redis will still be cleared by the handle_settings_change in the signals.py file https://github.com/ansible/awx/blob/devel/awx/conf/signals.py. 

I would assume with 3 tasks pods it would send pg_notify 3 times. Are the pods all receiving the same requests? Are we worried about duplication of work?

Do you mean removing the clear_settings_job and just calling pg_notify in on_commit?

Reply:
> if i have 3 task pod would this send out 3 pg_notify or 1?

@TheRealHaoLiu pg_notify does a NOTIFY to a "channel" in postgres, and every client that is LISTENing will receive the message. I am capitalizing to conform with postgres SQL syntax.

IMHO, the _best_ implementation would NOTIFY 1 channel when a setting is changed. You _could_ have a `tower_settings_change_web` channel and `tower_settings_change_task` channel, but I have never discussed that.

Assuming a single-channel implementation (because it is my preference)... if there are 3 task pods, then there will be 3 dispatcher processes, one for each pod, listening to the `tower_settings_change` channel. The NOTIFY happens once, but _postgres_ sends the message to _all listeners_, which would be 3 in this case.

> Do you mean removing the clear_settings_job and just calling pg_notify in on_commit?

To clarify what I would prefer, which is the same direction I was trying to give in my last comment - revert to put back in the prior code inside of `clear_setting_cache`, but keep the change where you changed its queue to `tower_settings_change`. Keep the channels you have defined for the dispatcher and other service to listen to.

Now, do some testing, and log / observe what messages the dispatcher service and the `run_clear_cache` service are getting. You should observe that they are _both_ getting the message to run this task. The only problem is that `run_clear_cache` isn't equipped to interpret this message. What I tried to say in prior comments is that we should look at that data that the current system (the "publisher" part of the dispatcher) sends in the message. Then `run_clear_cache` should parse the data to get the `setting_keys` and then directly (meaning, not with `.delay` or `.apply_async`) call `clear_setting_cache` with that argument.